packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -293,9 +293,11 @@ class _GameRenderObjectWidget extends LeafRenderObjectWidget {
 
   @override
   RenderBox createRenderObject(BuildContext context) {
-    return RenderConstrainedBox(
-      child: GameRenderBox(context, game),
-      additionalConstraints: const BoxConstraints.expand(),
-    );
+    return GameRenderBox(context, game);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, GameRenderBox renderObject) {
+    renderObject..game = game;
```

Comment:
What is it that happens here?

Reply:
Render objects are log living, while widgets live and die every rebuild. Since parameters of a widget can change between builds, we may need to update the render object with the current instance of game. As of today, we don't react to that change.